client-v2/src/main/java/com/clickhouse/client/api/internal/StopWatch.java

```diff
@@ -0,0 +1,41 @@
+package com.clickhouse.client.api.internal;
+
+import java.util.concurrent.TimeUnit;
+
+public class StopWatch {
+
+    long elapsedNanoTime = 0;
+    long startNanoTime;
+
+    public StopWatch() {
+        // do nothing
+    }
+
+    public StopWatch(long startNanoTime) {
+        this.startNanoTime = startNanoTime;
+    }
+
+    public void start() {
+        startNanoTime = System.nanoTime();

```

Comment:
from what i remember nano time is more expensive https://www.tutorialspoint.com/java-system-nanotime-vs-system-currenttimemillis

Reply:
I've found some good doc from JDK developer https://web.archive.org/web/20160308031939/https://blogs.oracle.com/dholmes/entry/inside_the_hotspot_vm_clocks 
It explains why nanoTime() *may* be slow and what should be used for precise measurement.  

Conclusion:
- nanoTime() may be slow in scale of microseconds
- nanoTime() may require hardware IO and that is why it may be slow
- nanoTime() is what recommended for elapsed time measurements. 
- currentTimeMillis() is not so precise and may have a lag of 10ms
- we will use nanoTime() until we see a noticeable drop in performance.